verifier/verifier.go

```diff
@@ -50,16 +54,25 @@ func NewFromConfig() (notation.Verifier, error) {
 
 // New creates a new verifier given trustPolicy, trustStore and pluginManager
 func New(trustPolicy *trustpolicy.Document, trustStore truststore.X509TrustStore, pluginManager plugin.Manager) (notation.Verifier, error) {
+	return NewWithRevocationClient(trustPolicy, trustStore, pluginManager, &http.Client{Timeout: 5 * time.Second})
+}
+
+// NewWithRevocationClient creates a new verifier given trustPolicy, trustStore, pluginManager, and revocationClient
+func NewWithRevocationClient(trustPolicy *trustpolicy.Document, trustStore truststore.X509TrustStore, pluginManager plugin.Manager, revocationClient *http.Client) (notation.Verifier, error) {
```

Comment:
Instead of passing `revocationClient *http.Client`, how about `revocationClient revocation.Revocation`. Doing so also helps the unit tests.

Reply:
I'm not sure that is the best approach. Since the creation of a revocation.Revocation object via New can result in an error, that means that an additional error check needs to be done everywhere that this would be switched. This would make the unit tests more complex, but more importantly, there would potentially be revocation related errors when calling the verify constructors (as it would need to call revocation.New with a client). I think that it makes sense to keep these errors confined to the revocation check (considering a use case like someone wanting to skip revocation, but getting a revocation related error anyways in a different spot)

Reply:
After discussing with Pritesh, I think this error here is okay as they can pass their own revocation.Revocation implementation to avoid any issues. I have added a test to ensure that the default implementation used in the New constructor does not result in errors in the future. Resolving